src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Handlers/PublishContentContext.cs

```diff
@@ -2,15 +2,18 @@ namespace OrchardCore.ContentManagement.Handlers
 {
     public class PublishContentContext : ContentContextBase
     {
-        public PublishContentContext(ContentItem contentItem, ContentItem previousContentItem) : base(contentItem)
+        public PublishContentContext(ContentItem contentItem, ContentItem previousContentItem, bool? isPublishing = true) : base(contentItem)
```

Comment:
why nullable `bool?` ?

Reply:
I will take a look at these comments @sebastienros 
I just wanted to have an optional param there so that we don't need to pass it every time.

Reply:
The new `IsPublishing` worries me, the DefaultContentManager mentions that something can be null and handlers should take that into account, not create new dynamic fields like this. It's like saying we have a thing (Publish) but it's not really for that (Publishing)

Reply:
And by that I mean on the DefaultContentManager.

Reply:
Though, I did not change the behavior of what it was doing before. So, I'm affecting the PublishingItem only when we are actually publishing. I'm just moving it to a param instead of leaving it as a public setter. Because, that's what we don't want, to let people affect this PublishingItem other than on instanciation.

Reply:
That or we should have an Unpublishing event